src/storage/store/column_chunk_data.cpp

```diff
@@ -28,8 +29,13 @@ namespace storage {
 ColumnChunkMetadata uncompressedFlushBuffer(const uint8_t* buffer, uint64_t bufferSize,
     BMFileHandle* dataFH, page_idx_t startPageIdx, const ColumnChunkMetadata& metadata) {
     KU_ASSERT(dataFH->getNumPages() >= startPageIdx + metadata.numPages);
-    dataFH->getFileInfo()->writeFile(buffer, bufferSize,
-        startPageIdx * BufferPoolConstants::PAGE_4KB_SIZE);
+    if (dataFH->isInMemoryMode()) {
+        const auto frame = dataFH->getBM()->getFrame(*dataFH, startPageIdx);
+        memcpy(frame, buffer, bufferSize);
+    } else {
+        dataFH->getFileInfo()->writeFile(buffer, bufferSize,
```

Comment:
Maybe we should move away from exposing the FileInfo through the file handle and add a write function that more or less is equivalent to this block (i.e. memcpy in in-memory-mode and use the FileInfo otherwise).

Reply:
Separating this to another PR for refactoring.